src/workerd/api/http.c++

```diff
@@ -1121,8 +1122,13 @@ jsg::Ref<Request> Request::clone(jsg::Lock& js) {
   auto cfClone = cf.deepClone(js);
   auto bodyClone = Body::clone(js);
 
-  return jsg::alloc<Request>(method, url, redirect, kj::mv(headersClone), getFetcher(), getSignal(),
-      kj::mv(cfClone), kj::mv(bodyClone));
+  return jsg::alloc<Request>(method, url, redirect, kj::mv(headersClone), getFetcher(),
+      /* signal */ getThisSignal(), kj::mv(cfClone), kj::mv(bodyClone), /* thisSignal */ kj::none);
```

Comment:
I'm treating `req.clone()` and `new Request(req)` differently in that former always clones the associated signal, while the latter will respect `ignoreForSubrequests` and ignore the signal of the incoming fetch. Let me know if that doesn't sound right.

Reply:
Hmm, this is not the solution I was expecting. I thought that the implementation of `fetch()` would be modified to ignore the signal if it came from the original top-level request. Why should cloning the request (whether via `.clone()` or the constructor) drop the signal?

Reply:
Actually if someone just does `fetch(originalRequest)`, the request is not cloned at all, so doesn't this not even solve that case?

Reply:
It does work because fetch calls the Request constructor again, but yeah this isn't the right way to accomplish this. I fixed it.